include/jemalloc/internal/arena_inlines_b.h

```diff
@@ -163,62 +162,110 @@ arena_vsalloc(tsdn_t *tsdn, const void *ptr) {
 }
 
 JEMALLOC_ALWAYS_INLINE void
-arena_dalloc(tsdn_t *tsdn, extent_t *extent, void *ptr, tcache_t *tcache,
-    bool slow_path) {
+arena_dalloc(tsdn_t *tsdn, void *ptr, tcache_t *tcache, bool slow_path) {
 	assert(!tsdn_null(tsdn) || tcache == NULL);
 	assert(ptr != NULL);
 
-	szind_t szind = extent_szind_get(extent);
-	if (likely(extent_slab_get(extent))) {
+	rtree_ctx_t rtree_ctx_fallback;
+	rtree_ctx_t *rtree_ctx = tsdn_rtree_ctx(tsdn, &rtree_ctx_fallback);
```

Comment:
This is implemented in https://github.com/jemalloc/jemalloc/pull/709/commits/23b986944042cc9cc327f00fe2ce903753093b2e.